code/game/objects/structures/mirror.dm

```diff
@@ -92,6 +95,47 @@
 		if(BURN)
 			playsound(src, 'sound/effects/hit_on_shattered_glass.ogg', 70, 1)
 
+// mirror frames
+
+/obj/item/wallframe/mirror
+	name = "mirror frame"
+	desc = "Really frames your face."
+	icon = 'icons/obj/wallframe.dmi'
+	icon_state = "mirror"
+	pixel_shift = 32
+	inverse = TRUE
+	result_path = /obj/structure/mirror_construct
+
+/obj/structure/mirror_construct
+	name = "mounted mirror frame"
+	desc = "A mirror without the mirror. Ideal for the offices of clowns, nonhumans, and other unattractive crew members."
+	icon = 'icons/obj/wallframe.dmi'
+	icon_state = "mirror"
+	anchored = TRUE
+	layer = WALL_OBJ_LAYER
+	max_integrity = 200
+	var/obj/structure/newmirror = null
+	armor = list("melee" = 50, "bullet" = 10, "laser" = 10, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 50)
+
+/obj/structure/mirror_construct/attackby(obj/item/W, mob/user, params)
+	add_fingerprint(user)
+	if(istype(W, /obj/item/stack/sheet/glass))
+		if(!W.tool_start_check(user, amount=2))
+			return
+		playsound(src, 'sound/items/deconstruct.ogg', 50, 1)
+		to_chat(user, "<span class='notice'>You start to add glass to the mirror frame...</span>")
+		if(W.use_tool(src, user, 20, amount=2))
+			to_chat(user, "<span class='notice'>You put in the glass panel.</span>")
+			newmirror = new /obj/structure/mirror(loc)
+			if(pixel_x)

```

Comment:
Pretty sure this if isn't actually needed

Reply:
on attach() it can be x or y dependent on the dir

Reply:
Yes, but setting both of them has the same effect - consider the case where pixel_x = 0, so we go to the else branch. not setting pixel_x has the exact same effect as setting it to 0, so it makes no difference.

Reply:
but if the pixel shift is on pixel_y something has to account for that